### PR TITLE
Set a new payment status after an intent was created

### PR DESCRIPTION
**Pull request by torotil**

### CODE REVIEW

File: src/StripeController.php [hunk 1]
```diff
@@ -233,6 +233,7 @@ public function ajaxCallback(\Payment $payment, array $form, array $form_state)
         'stripe_id' => $intent->id,
         'type' => $intent->object,
       ];
+      $payment->setStatus(new \PaymentStatusItem(STRIPE_PAYMENT_STATUS_INTENT_CREATED));
```
Comment (katzmo) [inline src/StripeController.php]: This does not catch intents that come as part of a subscription. Do we want a status change for them as well? But not all subscriptions use intents (only if confirmation is required).

Comment (torotil) [reply src/StripeController.php]: My aim is that all payments get a PENDING status before the first success. I would do so for setup intents as well, yes.

Comment (katzmo) [reply src/StripeController.php]: And for subscriptions without intent? They don’t get a pending state, but also they aren’t pending… Still, for the logcrm events to work properly they'd need one to also send a success status.

Comment (katzmo) [reply src/StripeController.php]: https://github.com/moreonion/campaignion_logcrm/pull/23 now always sends success states.

**Review by katzmo — APPROVED** [review]

**Pull request merged**